pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -583,6 +533,19 @@ func podSetUpdates(wl *kueue.Workload, pr *autoscaling.ProvisioningRequest) []ku
 	})
 }
 
+func (c *Controller) deactivateWorkload(ctx context.Context, wl *kueue.Workload, msg string) error {
+	updatedWl := &kueue.Workload{}
+	if err := c.client.Get(ctx, client.ObjectKeyFromObject(wl), updatedWl); err != nil {
+		return err
+	}
+	workload.SetEvictedCondition(updatedWl, kueue.WorkloadEvictedByDeactivation, msg)
+	if err := workload.ApplyAdmissionStatus(ctx, c.client, updatedWl, true); err != nil {
+		return err
+	}
+	updatedWl.Spec.Active = ptr.To(false)
```

Comment:
The problem for the reverse engineering is that the logic needs to know that the ProvReq has the CapacityRevoked condition. And the workload controller should not know about ProvReqs, due to separation of concerns. My recommendation to @PBundyra was that we should just rely on the Rejected status for the admission checks. However, right now this is limited just to the case when ProvReqs have WorkloadInactive, and not other Rejects (like reaching the maximum number of failures).

I think we should change the logic to always set the workload to inactive when the admission check is in Reject and the workload is not admitted.
But I'm undecided whether we should leave it for a follow up.

I think it might be safer to do it as a preparatory PR. I'm tempted to leave these two PRs for v0.8 not to delay v0.7 any further.

Reply:
I agree, in this PR we can just set `active: false` and rely on workload controller to set `Evicted: true` with "WorkloadInactive" reason, and generic message.

Reply:
@alculquicondor I think going forward (0.8+) a viable approach could be to have a DeactivationTarget condition. It would be set before `active: false`. Then, its message could be used in a generic fashion for `Evicted` condition. A similar approach I menion here: https://github.com/kubernetes-sigs/kueue/issues/2291#issuecomment-2135185436